contracts/VToken.sol

```diff
@@ -774,6 +774,10 @@ contract VToken is VTokenInterface, Exponential, TokenErrorReporter {
         // EFFECTS & INTERACTIONS
         // (No safe failures beyond this point)
 
+        /* We write previously calculated values into storage */
+        totalSupply = vars.totalSupplyNew;
+        accountTokens[redeemer] = vars.accountTokensNew;
```

Comment:
If we move state changes here, we can't `return failOpaque(...)` anymore, we can only revert. Can you please replace all the error conditions to reverts after the state changes are made?

Reply:
But I don't get this. As the state changes are just moved above the transfer and there is no check of state anywhere down. so it will not effect the case.

Reply:
@kkirka any update?

Reply:
But I don't get this. As the state changes are just moved above the transfer and there is no check of state anywhere down. so it will not effect the case.